### PR TITLE
Allow custom User Agent for all API invocations

### PR DESCRIPTION
**Pull request by drhoden-fastly**
### TL;DR

This PR implements the custom user agent request initiated by @phamann from https://github.com/peterbourgon/fastly-exporter/issues/90

### Why?

`http.RoundTripper` was selected to drive this solution because it was very obvious, configurable and is similar to the approach taken with `rt.WithUserAgent` injecting the user agent at runtime. 

@peterbourgon  suggested baking the user agent at build time. This was not explored as it is unclear if this meant adding to the linker Flag invocation in the Makefile or something else. Guidance is appreciated and I am happy to explore this solution or any mods to this PR (reference: https://github.com/peterbourgon/fastly-exporter/issues/90#issuecomment-1008150330).

👋🏿 @peterbourgon hope you are well and thanks for the review :) 

### Note:

n/a

### CODE REVIEW

**Comment by peterbourgon** [comment]
Thanks! I think this feature request moves the api.ServiceCache, api.DatacenterCache, rt.Manager, and rt.Subscriber closer together in certain respects, and reveals that they should probably be refactored a little bit. Let me do that real fast and I'll cc: you for a review.

**Comment by drhoden-fastly** [comment]
neat thanks @peterbourgon. If I can clean up anything please lmk.

**Comment by drhoden-fastly** [comment]
I am going to be stepping away for a bit, but I chatted with @phamann directly and he agreed to address any feedback in my absence. Thanks again for your work!
